lib/src/rules/curly_braces_in_flow_control_structures.dart

```diff
@@ -100,7 +100,9 @@ class _Visitor extends SimpleAstVisitor {
       }
     } else {
       _check(node.thenStatement);
-      if (elseStatement is! IfStatement) {
+      if (elseStatement is IfStatement) {
+        _check(elseStatement.thenStatement);
```

Comment:
While I agree that the test case you added should be flagged, I think that the impact of this change would be to cause two lints to be generated for the following code:
```dart
void f() {
  if (true) {
  } else if (true)
    return;
}
```
One when visiting the outer `if` and `_check` discovers that the argument isn't a block, and one when visiting the inner `if` and we get to line 99. (I haven't applied your delta to confirm, so I might be wrong.)

If I'm right, then I believe the correct fix would be to add code between lines 91 and 92 that will `_check` the `thenStatement` if the `node` is the `elseStatement` of an outer `if` (and return to skip other checks).

Reply:
Oh, that's interesting.

I tested with both

```dart
testIfElse() {
  void f() {
    if (true) {
    } else if (true) return;
  }
}
```

and

```dart
void f() {
  if (true) {
  } else if (true)
    return;
}
```

And lint is triggered only once.


BTW, without the proposed change, the below doesn't trigger lint:

```dart
  void f() {
    if (true) {
    } else if (true) return;
  }
```

Reply:
Apologies, you are totally right.